common/ssl_calls.c

```diff
@@ -592,6 +647,15 @@ ssl_tls_accept(struct ssl_tls *self, long ssl_protocols,
                      SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER |
                      SSL_MODE_ENABLE_PARTIAL_WRITE);
     SSL_CTX_set_options(self->ctx, options);
+    DH *dh = get_dh2236 ();
+    if (SSL_CTX_set_tmp_dh(self->ctx, dh) != 1) {
+        g_writeln("SSL_CTX_set_tmp_dh failed");
+        return 1;
```

Comment:
Potential memory leak here, the `SSL_CTX` object is never freed afterwards.

Reply:
ok, my bad, its is freed afterwards by `ssl_tls_delete()` actually.

Reply:
can we use `SSL_set_dh_auto()` here like in `ecdh`?

Reply:
```SSL_set_dh_auto()``` can be used with openssl 1.1.0 and newer. But there is the cons of having two code paths, which makes testing and reproducing issues harder.I'm 50-50 on this choice to be honest